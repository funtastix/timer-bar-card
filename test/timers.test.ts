import { multiply, HomeAssistant, PlaywrightBrowser, PlaywrightElement } from "hass-taste-test";
import { synchronizeTimerPaused, toMatchDualSnapshot } from "./util";
expect.extend({ toMatchDualSnapshot });

const CONFIGURATION_YAML = `
timer:
  ${multiply(10, (i) => `
  test${i}:
    duration: "00:${i}:00"
`)}
`;

let hass: HomeAssistant<PlaywrightElement>;

beforeAll(async () => {
  hass = await HomeAssistant.create(CONFIGURATION_YAML, {
    browser: new PlaywrightBrowser(process.env.BROWSER || "firefox"),
  });
  await hass.addResource(__dirname + "/../dist/timer-bar-card.js", "module");
}, 30000);
afterAll(async () => await hass.close());

it("Entity Row Styles", async () => {
  const dashboard = await hass.Dashboard([
    { type: "custom:timer-bar-card", entity: "timer.test1" },
  ]);
  const card = dashboard.cards[0];
  await expect(card).toMatchDualSnapshot("idle");

  await hass.callService("timer", "start", {}, { entity_id: "timer.test1" });
  await expect(card).toMatchDualSnapshot("running");

  await synchronizeTimerPaused(hass, "timer.test1", "00:00:58");
  await hass.callService("timer", "pause", {}, { entity_id: "timer.test1" });
  await expect(card).toMatchDualSnapshot("paused");

});

it("Card UI Styles", async () => {
  const dashboard = await hass.Dashboard([
    { type: "custom:timer-bar-card", entities: ["timer.test2", "timer.test3"] },
  ]);
  const card = dashboard.cards[0];
  await expect(card).toMatchDualSnapshot("idle");

  await hass.callService("timer", "start", {}, { entity_id: "timer.test2" });
  await expect(card).toMatchDualSnapshot("running");

  await synchronizeTimerPaused(hass, "timer.test2", "00:01:58");
  await hass.callService("timer", "pause", {}, { entity_id: "timer.test2" });
  await expect(card).toMatchDualSnapshot("paused");
});

it("Debug window appears", async () => {
  const dashboard = await hass.Dashboard([
    { type: "custom:timer-bar-card", entities: ["timer.test4", "timer.test5"], debug: true },
  ]);
  const element = await dashboard.cards[0].element()
  expect(await element.$$eval("code", (els) => els.length)).toBe(2)
});
