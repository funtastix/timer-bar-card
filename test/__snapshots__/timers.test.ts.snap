// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card UI Styles: idle 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card outlined=\\"false\\">
    <slot></slot>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-icon data-state=\\"idle\\" icon=\\"hass:timer-outline\\">…</ha-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\">
            test2
          </div>
          <slot></slot>
          <div class=\\"text-content\\">
            Idle
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-icon data-state=\\"idle\\" icon=\\"hass:timer-outline\\">…</ha-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\">
            test3
          </div>
          <slot></slot>
          <div class=\\"text-content\\">
            Idle
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;

exports[`Card UI Styles: paused 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card outlined=\\"false\\">
    <slot></slot>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-icon data-state=\\"paused\\" icon=\\"hass:timer-outline\\">…</ha-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\">
            test2
          </div>
          <slot></slot>
          <div class=\\"status\\">
            Paused
          </div>
          <div class=\\"text-content\\">
            1:58
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-icon data-state=\\"idle\\" icon=\\"hass:timer-outline\\">…</ha-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\">
            test3
          </div>
          <slot></slot>
          <div class=\\"text-content\\">
            Idle
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;

exports[`Card UI Styles: running 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <ha-card outlined=\\"false\\">
    <slot></slot>
    <div class=\\"card-content\\">
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-icon data-state=\\"active\\" icon=\\"hass:timer-outline\\">…</ha-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\">
            test2
          </div>
          <slot></slot>
          <div class=\\"bar-container\\">
            <div class=\\"bar\\">
              <div>
              </div>
            </div>
          </div>
          <div class=\\"text-content\\">
            1:59
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
      <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
        <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
          <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
            <ha-icon data-state=\\"idle\\" icon=\\"hass:timer-outline\\">…</ha-icon>
          </state-badge>
          <div class=\\"info pointer text-content\\">
            test3
          </div>
          <slot></slot>
          <div class=\\"text-content\\">
            Idle
          </div>
        </hui-generic-entity-row>
      </timer-bar-entity-row>
    </div>
  </ha-card>
</timer-bar-card>
"
`;

exports[`Entity Row Styles: idle 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
    <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
      <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
        <ha-icon data-state=\\"idle\\" icon=\\"hass:timer-outline\\">…</ha-icon>
      </state-badge>
      <div class=\\"info pointer text-content\\">
        test1
      </div>
      <slot></slot>
      <div class=\\"text-content\\">
        Idle
      </div>
    </hui-generic-entity-row>
  </timer-bar-entity-row>
</timer-bar-card>
"
`;

exports[`Entity Row Styles: paused 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
    <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
      <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
        <ha-icon data-state=\\"paused\\" icon=\\"hass:timer-outline\\">…</ha-icon>
      </state-badge>
      <div class=\\"info pointer text-content\\">
        test1
      </div>
      <slot></slot>
      <div class=\\"status\\">
        Paused
      </div>
      <div class=\\"text-content\\">
        58
      </div>
    </hui-generic-entity-row>
  </timer-bar-entity-row>
</timer-bar-card>
"
`;

exports[`Entity Row Styles: running 1`] = `
"<timer-bar-card hass=\\"[object Object]\\">
  <timer-bar-entity-row config=\\"[object Object]\\" hass=\\"[object Object]\\">
    <hui-generic-entity-row config=\\"[object Object]\\" no-secondary=\\"\\">
      <state-badge _showIcon=\\"true\\" class=\\"pointer\\" stateObj=\\"[object Object]\\" tabindex=\\"0\\">
        <ha-icon data-state=\\"active\\" icon=\\"hass:timer-outline\\">…</ha-icon>
      </state-badge>
      <div class=\\"info pointer text-content\\">
        test1
      </div>
      <slot></slot>
      <div class=\\"bar-container\\">
        <div class=\\"bar\\">
          <div>
          </div>
        </div>
      </div>
      <div class=\\"text-content\\">
        59
      </div>
    </hui-generic-entity-row>
  </timer-bar-entity-row>
</timer-bar-card>
"
`;
